spanner/client.go

```diff
@@ -280,6 +281,10 @@ func (c *Client) Single() *ReadOnlyTransaction {
 		t.sh = sh
 		return nil
 	}
+	if c.sc != nil && c.ct == nil {
+		setCommonTags(c)
+	}
```

Comment:
This is not necessary here (and also not in the other methods that start a transaction). Instead, this initialization should be done in the `NewClientWithConfig` method, right after the `sessionClient` has been created and before returning the new client.